src/client/pythonEnvironments/discovery/locators/services/globalVirtualEnvronmentLocator.ts

```diff
@@ -168,15 +172,20 @@ export class GlobalVirtualEnvironmentLocator extends Locator {
         return undefined;
     }
 
-    private async registerWatchers(): Promise<void> {
+    private async startWatchers(): Promise<void> {
         const dirs = await getGlobalVirtualEnvDirs();
-        dirs.forEach((d) => watchLocationForPythonBinaries(d, async (type: FileChangeType, executablePath: string) => {
-            // Note detecting kind of virtual env depends on the file structure around the executable, so we need to
-            // wait before attempting to detect it. However even if the type detected is incorrect, it doesn't do any
-            // practical harm as kinds in this locator are used in the same way (same activation commands etc.)
-            await sleep(1000);
-            const kind = await getVirtualEnvKind(executablePath);
-            this.emitter.fire({ type, kind });
-        }));
+        dirs.forEach(
+            (d) => this.disposables.push(
+                watchLocationForPythonBinaries(d, async (type: FileChangeType, executablePath: string) => {
+                    // Note detecting kind of virtual env depends on the file structure around the
+                    // executable, so we need to wait before attempting to detect it. However even
+                    // if the type detected is incorrect, it doesn't do any practical harm as kinds
+                    // in this locator are used in the same way (same activation commands etc.)
+                    await sleep(1000);
```

Comment:
This should only happen for creation, right?

```suggestion
                    if (type === FileChangeType.Created) {
                        // Note detecting kind of virtual env depends on the file structure around the
                        // executable, so we need to wait before attempting to detect it. However even
                        // if the type detected is incorrect, it doesn't do any practical harm as kinds
                        // in this locator are used in the same way (same activation commands etc.)
                        await sleep(1000);
                    }
```

Reply:
For creation and maybe also modification? Although modification shouldn't change the kind.

Reply:
Did it only for creation assuming modification generally doesn't change the type.